### PR TITLE
MOD-4822: Disable recursion limit on RDB load

### PR DESCRIPTION
**Pull request by gkorland**

### CODE REVIEW

**Review by MeirShpilraien — COMMENTED** [review]
Should we add a test?
We will need to do it for hdt as well.

File: tests/pytest/test.py [hunk 1]
```diff
@@ -1279,6 +1279,20 @@ def testFilterPrecedence(env):
     r.assertEqual(json.loads(res), [doc[0]])
     r.expect('JSON.GET', 'doc', '$[?(@.f==true || @.one==1 && @.t==false)]').equal('[]')
 
+def testRDBUnboundedDepth(env):
+    # Test RDB Unbounded Depth load
+    r = env
+    json_126 = '{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"a":{"z":"a"}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}}'
```
Comment (MeirShpilraien) [inline tests/pytest/test.py:1285]: Maybe its better to generate it?
Not critical..

File: tests/pytest/test.py [hunk 2]
```diff
@@ -1279,6 +1279,20 @@ def testFilterPrecedence(env):
...
+    r.expect('JSON.SET', 'doc', '$', json_126).ok()
+
+    # concat the string_126 at the end of itself
+    r.expect('JSON.SET', 'doc', '$..z', json_126).ok()
+    
+    # RDB dump and restore the key 'doc' and check that the key is still valid
+    dump = env.execute_command('dump', 'doc', **{NEVER_DECODE: []})
+    r.expect('RESTORE', 'doc1', 0, dump).ok()
+    r.expect('JSON.GET', 'doc1', '$..z..z').equal('["a"]')
```
Comment (MeirShpilraien) [inline tests/pytest/test.py:1294]: Should we also check `debug reload`?

**Review by MeirShpilraien — DISMISSED** [review]
:+1:

File: tests/pytest/test.py (hunk 1 above)
Comment (oshadmi) [reply tests/pytest/test.py:1285]: Can use https://github.com/RedisJSON/RedisJSON/blob/7d91ab35afa743a9a007e49fa9d2d6070f77fd80/tests/pytest/test.py#L1159

**Pull request merged**